dandischema/tests/test_models.py

```diff
@@ -611,7 +622,7 @@ def test_resource() -> None:
 
 
 def test_basetype() -> None:
-    props = json.loads(BaseType.schema_json())["properties"]
+    props = json.loads(json.dumps(BaseType.model_json_schema()))["properties"]
```

Comment:
if `model_json_schema` returns a dict, then there is no need to load from a dump.

`schema_json` returned a string in the past.

Reply:
@satra 

> `schema_json` returned a string in the past.

Right, but it appears that v1 of Pydantic had a `schema()` method, which didn't return a string.

Reply:
The only possible reason I see is that to test the dumping and reloading since the line is in a test.

But at this point, I take it as decided that we have no reason to do dumping and reloading, and I will remove the redundancy.

Reply:
`schema_json` ran through the schema update code in the base model, i can't remember if `schema` did that.

Reply:
According to the documentation of [`BaseModel.model_json_schema`](https://docs.pydantic.dev/latest/api/base_model/#pydantic.main.BaseModel.model_json_schema), its return is guaranteed to be JSONable. So, I will just eliminate the unnecessary serialization to a string at this point, as John suggested by using `BaseModel.model_json_schema`.

Reply:
as long as it calls your `post_process_json_schema` function before returning the value. if it doesn't then it's returning the wrong schema.

Reply:
Yes, `post_process_json_schema` is called each time the JSON schema for the `DandiBaseModel` or its sub-models is generated, regardless of which interface is used to initiated the generation (at least this is true for Pydantic V2).

`post_process_json_schema` is used as the `json_schema_extra` value in `model_config` which r[eplaces `schema_extra` in the `Config` class in Pydantic V1](https://docs.pydantic.dev/latest/migration/#changes-to-config).